pennylane_lightning/src/simulators/lightning_kokkos/algorithms/tests/Test_AdjointJacobianKokkos.cpp

```diff
@@ -0,0 +1,503 @@
+#include <algorithm>
+#include <cmath>
+#include <complex>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "AdjointJacobianKokkos.hpp"
+#include "StateVectorKokkos.hpp"
+#include "TestHelpers.hpp"
+#include "Util.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::Util;
+using namespace Pennylane::LightningKokkos::Algorithms;
+} // namespace
+/// @endcond
+
+/**
+ * @brief Tests the constructability of the AdjointDiff.hpp classes.
+ *
+ */
+TEMPLATE_TEST_CASE("AdjointJacobian::AdjointJacobian", "[AdjointJacobian]",
+                   float, double) {
+    SECTION("AdjointJacobian<TestType> {}") {
+        REQUIRE(std::is_constructible<
+                AdjointJacobian<StateVectorKokkos<TestType>>>::value);
+    }
+}
+
+TEST_CASE("AdjointJacobian::AdjointJacobian Op=RX, Obs=Z",
+          "[AdjointJacobian]") {
```

Comment:
Some of the tests we have in this file (or very similar ones) are present in the agnostic set of tests.
If you think the tests here, given LKokkos specificities, can be portable to the agnostic level, this would be great.
If needed, feel free to expand the set of agnostic tests.

Reply:
You are right, the difference is (for example) the call
```
adj.adjointJacobian(psi, jacobian, {obs1, obs2}, ops, {0}, true);
```
for the specialized class and 
```
adj.adjointJacobian(std::span{jacobian}, tape, true);
```
for the agnostic class. Since the bindings depend on the first, I would like to first add bindings, then test the bindings, then mutate the bindings and finally remove the old interface and tests.

Reply:
At the end of the day, we want LKokkos adjointJacobian (and any other method) to have the same interface as LQubit methods. As a matter of fact, we want it to all backends. So, to address your concern with bindings, I suggest you add the desired interface (same as LQubit) and keep the old one for now (just for bindings). This way we isolate as most as possible of the C++ work on this PR, and keep the bindings PR just for bindings (including removing the C++ interface for bindings).

Reply:
Agreed, so for now both interfaces exist and are tested.